WordPress/Classes/Models/AbstractPost.m

```diff
@@ -452,8 +452,17 @@ - (BOOL)isDraft
 
 - (void)publishImmediately
 {
-    self.dateModified = [NSDate date];
-    [self setDateCreated:self.dateModified];
+    // If the post was modified later than it was created
+    // then this is a post that was published at some point
+    // we will not update the dateCreated
```

Comment:
So are you saying that we should remove the Publish Immediately option when a Post has already been Published? (which we are assuming happened when `[self.dateCreated earlierDate:self.dateModified] == self.dateCreated`) I think the `metadata` field can help us here, there is a key called `_rest_api_published` so we could use that (undocumented I know). Do you know if that would be set for self hosted sites?